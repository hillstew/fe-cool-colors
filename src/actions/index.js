export const setColors = colors => ({
  type: 'SET_COLORS',
  colors
});

export const setLoading = bool => ({
  type: 'SET_LOADING',
  bool
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setProjects = projects => ({
  type: 'SET_PROJECTS',
  projects
});

export const setPalettes = palettes => ({
  type: 'SET_PALETTES',
  palettes
});

export const toggleLocked = id => ({
  type: 'TOGGLE_LOCKED',
  id
});
