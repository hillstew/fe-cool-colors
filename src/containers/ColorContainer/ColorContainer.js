import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ColorCard } from '../../components/ColorCard/ColorCard';
import { toggleLocked, setColors } from '../../actions';
import { updateColors } from '../../utils/helper';

export class ColorContainer extends Component {
  render() {
    const { colors, toggleLocked, setColors } = this.props;
    const colorBoxes = colors.map(color => {
      return <ColorCard
        key={color.id}
        color={color}
        toggleLocked={toggleLocked} />;
    });

    return <div className="ColorContainer">
      {colorBoxes}
      <div>
        <button 
          className='generate-color-button'
          onClick={() => setColors(updateColors(colors))}>Generate!</button>
      </div>
    </div>;
  }
}

export const mapStateToProps = state => ({
  colors: state.colors
});

export const mapDispatchToProps = dispatch => ({
  toggleLocked: id => dispatch(toggleLocked(id)),
  setColors: colors => dispatch(setColors(colors))
});

export default connect(mapStateToProps, mapDispatchToProps)(ColorContainer);
