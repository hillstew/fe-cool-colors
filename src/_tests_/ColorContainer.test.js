import React from 'react';
import { shallow } from 'enzyme';
import { ColorContainer } from '../containers/ColorContainer/ColorContainer';
import { toggleLocked, setColors } from '../actions';

describe('ColorContainer', () => {
  let wrapper;
  const mockProps = {
    colors: ['red', 'orange', 'yellow', 'green', 'blue'],
    toggleLocked: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<ColorContainer {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
