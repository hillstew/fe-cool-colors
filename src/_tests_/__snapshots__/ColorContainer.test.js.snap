// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorContainer should match the snapshot 1`] = `
<div
  className="ColorContainer"
>
  <ColorCard
    color="red"
    toggleLocked={[MockFunction]}
  />
  <ColorCard
    color="orange"
    toggleLocked={[MockFunction]}
  />
  <ColorCard
    color="yellow"
    toggleLocked={[MockFunction]}
  />
  <ColorCard
    color="green"
    toggleLocked={[MockFunction]}
  />
  <ColorCard
    color="blue"
    toggleLocked={[MockFunction]}
  />
  <Connect(Controls) />
</div>
`;
